### ISSUE TITLE
Accessing Queue on Two Different Machines

### ISSUE THREAD

**Issue by InzamamAnwar** [peter-wangxu/persist-queue#148]
Hi, 
Persist-queue is an elegant solution for disk based queuing and easy to use as well. 

Now, my idea was to use a shared folder between two machines and put the queue in that folder. On one machine I am populating the queue in one process and consuming it in another process, of course setting `multithreading=True`. But When I access the queue from the other machine where I only want to consume/get values it throws error on either machine. The error is given below. 

Is there any possibility to implement this functionality? Or any other idea to accomplish this? Suggestions are sincerely welcomed

```
[2020-12-24 22:33:39,747: ERROR/ForkPoolWorker-1] Task app_queue.call_get_faces_from_video_async[4fa8578a-2262-4add-b390-6b72888e4a1c] raised unexpected: OperationalError('disk I/O error')
Traceback (most recent call last):
  File "/home/inzamam/anaconda3/envs/tf2/lib/python3.7/site-packages/celery/app/trace.py", line 409, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/inzamam/anaconda3/envs/tf2/lib/python3.7/site-packages/celery/app/trace.py", line 701, in __protected_call__
    return self.run(*args, **kwargs)
  File "/home/inzamam/Projects/face-detection-api/app_queue.py", line 96, in call_get_faces_from_video_async
    request_data = request_queue.get()
  File "/home/inzamam/anaconda3/envs/tf2/lib/python3.7/site-packages/persistqueue/sqlqueue.py", line 83, in get
    serialized = self._pop()
  File "/home/inzamam/anaconda3/envs/tf2/lib/python3.7/site-packages/persistqueue/sqlqueue.py", line 60, in _pop
    row = self._select()
  File "/home/inzamam/anaconda3/envs/tf2/lib/python3.7/site-packages/persistqueue/sqlbase.py", line 154, in _select
    return self._getter.execute(self._sql_select, args).fetchone()
sqlite3.OperationalError: disk I/O error

```

**Comment by issahammoud** [comment]
I have the same need as well! 
In fact, I was looking for a disk-based thread-safe way to communicate between different machines on shared disk. 
It was strange that this feature is not supported by default until I paid attention to "...stored in disk IN CASE OF ANY CRASH".
Do you know about a solution for this problem, other than creating lock on the disk?

**Comment by peter-wangxu** [comment]
This feature requires an external database like mysql or postgresql.
can this be fulfilled in your environment.

Thanks
Peter

**Comment by issahammoud** [comment]
I want to have the same functionality as described above. I want to run 1 worker per machine to put data in the queue (on nfs), then the master process (machine) will get the data.
I can't see why this cannot be accomplished with SQLite? 

Thank you for the great work by the way.

**Comment by InzamamAnwar** [comment]
@peter-wangxu by external database you mean separate sqlserver running somewhere and accessible by IP? Than yes It can be incorporated. 

@issahammoud when I experimented it was not possible with existing implementation. If you are able to run and can share it than please do contribute.

**Comment by issahammoud** [comment]
@peter-wangxu @InzamamAnwar 
It is my first time using SQLite, so I am not the expert of the topic. 
After a quick search on the Internet, I found the source of the problem, which is the journal_mode setted to WAL (Write-Ahead Logging). 
As they claim (see https://www.sqlite.org/wal.html): "All processes using a database must be on the same host computer; WAL does not work over a network filesystem."
So in order to use this library's version on several hosts, one should do the following:

from persistqueue import SQLiteQueue
queue = SQLiteQueue("your/path")
queue._conn.execute('PRAGMA journal_mode=OFF;')

You can choose other jounal_mode as described in the doc (https://www.sqlite.org/pragma.html#pragma_journal_mode).
However, I don't have an idea about the drop in performance, as WAL is faster.

**Comment by peter-wangxu** [comment]
This should be accomplished by feature:https://github.com/peter-wangxu/persist-queue/pull/174
available for version 0.8.0-alpha
